app/models/keyword.rb

```diff
@@ -4,6 +4,10 @@ class Keyword < ApplicationRecord
   enum status: { pending: 'pending', processing: 'processing', succeeded: 'succeeded', failed: 'failed' }
 
   has_many :keyword_search_entries, dependent: :destroy
+  has_one :keyword_stats, dependent: :nullify
+
+  delegate :top_ads_urls, :non_ads_urls, to: :keyword_decorator
+  delegate :top_ads_count, :total_ads_count, :non_ads_count, to: :keyword_stats, allow_nil: true
```

Comment:
The current behaviour is: if the `keyword` doesn't have any search entries, `keyword_stats` would be nil. All delegates fields as a result would be nil as well

```ruby
# This keyword doesn't have any search entries
keyword = Keyword.find(1)

keyword.keyword_stats # nil

keyword.keyword_stats.top_ads_count # nil
```

The behavior I want in the above case is: `keyword.keyword_stats.top_ads_count` equals 0 instead of nil. How can I achieve this?

What I've tried so far:
1. Manually setup `top_ads_count` method:
```ruby
def top_ads_count
  return keyword_stats ? keyword_stats.top_ads_count : 0
end
```

2. In case `keyword_stats` is nil, fallback to an empty instance
```ruby
delegate ..., to: keyword_stats_with_fallback

def keyword_stats_with_fallback
  return keyword_stats || KeywordStats.new() # we need some mechanisms to populate fields here as 0, otherwise they're all nil
end
```

What is the preferred way or is there any other solutions?

Reply:
Technically, the API response only needs the value set to zero. At the model layer, having `nil` is not an issue per se. So how about handling the data formatting at the serializer level instead? (which act as a view logic layer)

Reply:
There's definitely no problem handling that at serializer/presenter layers. However, I think it would be best if the can handle that in the model/decorator layer, to simplify the interface.

But in this case, there seems to be no easy way to achieve this. So I'll go with your suggestion.